### DISCUSSION TITLE
2.0

### DISCUSSION THREAD

**Discussion by txubelaxu** [txubelaxu/esphome-jk-bms#23] [Announcements]
Previous version: all the params were READ only.
In this new version, a lot of parameters are READ & WRITE.

"Broadcast Changes to all BMSs" SWITCH. If it is "on" and you make a change on a BMS, the same change will send to all other BMSs in the network.
![image](https://github.com/user-attachments/assets/d07e3ca0-9f5f-4169-84d3-1a6a5429dde9)

**Full Changelog**: https://github.com/txubelaxu/esphome-jk-bms/compare/v1.00...2.0

<hr /><em>This discussion was created from the release <a href='https://github.com/txubelaxu/esphome-jk-bms/releases/tag/2.0'>2.0</a>.</em>

**Comment by x230-thinkpad** [comment]
perfect all is good

![image](https://github.com/user-attachments/assets/cbb6f394-4d31-415b-944c-f69300d4c810)

I hope for next update we can adjust for those parameter settings, it make us easy, no need to open bluetooth apps or connected bms's into the PC

![image](https://github.com/user-attachments/assets/2634ce7e-6d6b-455b-b465-5d8ab4c37ab2)

**Comment by txubelaxu** [comment]
Hi @Patrick197030100,

Please test changing this line with this value:

`external_components_source: github://txubelaxu/esphome-jk-bms@main`

**Reply by txubelaxu** [reply]
Hi @Patrick197030100,

A lot of sensors have changed their type. From "sensor" to "number". Please, take a look to this example: https://github.com/txubelaxu/esphome-jk-bms/blob/main/esp32-S3-example-jkpb-rs485_0master_2slaves.yaml

You have to move from "sensor:" to "number:"

**Comment by Patrick197030100** [comment]
hello    
![Screenshot 2024-08-17 at 08-33-29 ESPHome – Home Assistant](https://github.com/user-attachments/assets/923fbaae-756e-4619-92ec-61d23b753579)
![Screenshot 2024-08-17 at 08-27-53 ESPHome – Home Assistant](https://github.com/user-attachments/assets/76b6ad1a-a93c-4420-840e-2ea627a8017b)


can you help me txubelaxu please I have an atom s3 like and rs485 sku u094 I want to put your file 2.00 and still error sniffer I am a novice it is very hard to understand because you have to adapt your file in relation to my config thank you to you

**Comment by Imanol82** [comment]
Amazing. Thank you Txubelaxu.

With this new version, the esp32 works as master or we can work same as until now and work with a Master and Slave BMS?


On the other hand, I have use a Modbus-TCP gateways to comunicate my inverter and HA, and I'm really surprised by how well it works, much better than an esp32, very industrial and stable (I have installed a POE one and it is amazing). I'm using the Kellerza plugin for Deye which has been modified for this type of gateways.  

Do you want to do something similar with your software? :)

If you need a tester or any help I'm here. 


Eskerrikasko!!! Lan izugarria egin dezu...

Thank you very much

**Reply by Imanol82** [reply]
I've seen on the project that ESP must be master.

In this case, putting all BMSs as slave, the comunication between the first one, first slave, and the inverter works fine?

Thanks!!

**Comment by Imanol82** [comment]
Hello again.

I have installed this new version, 2.0, and almost everything seems to work fine. I get the following errors (I don't know if I've done something wrong)

- The SOC that the inverter reads is only that of the Master (now Slave 1) and does not average the 2 BMS (SLAVE 1 and SLAVE 2).
![image](https://github.com/user-attachments/assets/28b97f5b-6f62-4ebb-bb53-4348ed58de90)
![image](https://github.com/user-attachments/assets/7918ff17-9516-4b3c-beed-99d70799cf4b)

-And a small (stupid) detail, in the Dashboard, in the second BMS the titles are missing:

![image](https://github.com/user-attachments/assets/b6847521-1c7d-403b-accf-301656452b77)
![image](https://github.com/user-attachments/assets/aae0bf49-dd57-436a-b955-82f668971fda)


Thank you very much for your work.

**Reply by txubelaxu** [reply]
I have just changed the dashboard. Is it OK now?

Many Thanks!!

**Comment by x230-thinkpad** [comment]
Hi again, may somebody can help me,

I have problem with last connection, address 0x07 is always become unknown
![image](https://github.com/user-attachments/assets/555fc56a-5959-442b-a66f-427bc76498db)

meanwhile other bms good connection

**Reply by txubelaxu** [reply]
umm... Are other param values OK for bms6?

**Reply by x230-thinkpad** [reply]
> umm... Are other param values OK for bms6?

other param is ok, here I attached my yaml file, last 2 days if connect in parallel the values not update.

I used ESP32-S3 N8R8 there is any possibility hardware not good enough?

once I try only handle 1 bms (0x01) everything is good, values keep update

but if the bms connected in parallel (0x02), (0x03), (0x04) the esp like stop update the values

![image](https://github.com/user-attachments/assets/24de8ca9-d7c8-4940-8e73-acc1130fbeaa)

![image](https://github.com/user-attachments/assets/79f21d8c-1e81-4009-b2b0-d06bad9b1193)

[logs_esp-rs485-gw_logs.txt](https://github.com/user-attachments/files/16791775/logs_esp-rs485-gw_logs.txt)
[esp-rs485-gw_.zip](https://github.com/user-attachments/files/16791762/esp-rs485-gw_.zip)
[logs_esp-rs485-gw_run.txt](https://github.com/user-attachments/files/16792008/logs_esp-rs485-gw_run.txt)
